plugins/module_utils/consul.py

```diff
@@ -5,25 +5,93 @@
 # SPDX-License-Identifier: GPL-3.0-or-later
 
 from __future__ import absolute_import, division, print_function
-__metaclass__ = type
-
 
-def get_consul_url(configuration):
```

Comment:
This function is part of the public API of the collection and needs to stay until the next major release.

(Same for `get_auth_headers` and `handle_consul_response_error` below.)